source/retryable-writes/tests/supported-multi-statement-failure.yml

```diff
@@ -3,47 +3,110 @@ data:
+                            document: { _id: 2, x: 22 }
+                    -
                         name: "updateOne"
                         arguments:
+                            filter: { _id: 2 }
+                            update: { $inc: { x : 1 }}
+                    -
+                        name: "deleteOne"
+                        arguments:
                             filter: { _id: 1 }
+                options: { ordered: true }
+        outcome:
+            error: true
+            collection:
+                data:
+                    - { _id: 1, x: 11 }
+    -
+        description: "Second updateOne is never committed"
+        failPoint:
+            mode: { skip: 1 }
+            data: { failBeforeCommitExceptionCode: 1 }
+        operation:
+            name: "bulkWrite"
+            arguments:
+                requests:
+                    -
+                        name: "insertOne"
+                        arguments:
+                            document: { _id: 2, x: 22 }
+                    -
+                        name: "updateOne"
+                        arguments:
+                            filter: { _id: 2 }
                             update: { $inc: { x : 1 }}
                     -
+                        name: "deleteOne"
+                        arguments:
+                            filter: { _id: 1 }
+                options: { ordered: true }
+        outcome:
+            error: true
+            collection:
+                data:
+                    - { _id: 1, x: 11 }
+                    - { _id: 2, x: 22 }
+    -
+        description: "Third updateOne is never committed"
+        failPoint:
+            mode: { skip: 1 }
+            data: { failBeforeCommitExceptionCode: 1 }
+        operation:
+            name: "bulkWrite"
+            arguments:
+                requests:
+                    -
                         name: "insertOne"
                         arguments:
                             document: { _id: 2, x: 22 }
+                    -
+                        name: "updateOne"
+                        arguments:
+                            filter: { _id: 1 }
+                            update: { $inc: { x : 1 }}
+                    -
+                        name: "updateOne"
+                        arguments:
+                            filter: { _id: 2 }
+                            update: { $inc: { x : 1 }}
                 options: { ordered: true }
         outcome:
             error: true
-            # The first "update" command will be applied, despite the network
-            # error. The second "insert" command in the batch is never executed.
             collection:
                 data:
                     - { _id: 1, x: 12 }
+                    - { _id: 2, x: 22 }
     -
         description: "InsertMany fails after multiple network errors"
         failPoint:
-            times: 2
+            mode: { skip: 2 }
```

Comment:
I failed to get this to work as expected, using 3.6.0-rc0 and `skip`.  As this is a single operation skip: 2 doesn't make sense as the whole operation is applied.

I updated to: `mode: { skip: 0 }` and updated the data to be just the original data - eg: nothing is added to the collection.

Reply:
This makes sense given Jack's explanation here: https://jira.mongodb.org/browse/SPEC-966?focusedCommentId=1701197&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1701197

Multiple operations in a single update/delete command each trigger the commit failpoint but multiple documents in a single insert are counted as a single commit.

Reply:
That agrees with [this comment in SPEC-966](https://jira.mongodb.org/browse/SPEC-966?focusedCommentId=1701197&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-1701197):

> As for each statement from each command being counted separately for the skip and times options, this is true for updates and deletes, but not for batch inserts, which actually will only trigger the op observer once for the entire batch of documents, meaning the failpoint will only be reached once.

I'll update the test to use `{ times: 2 }`.

Reply:
Note that we will ultimately have a test for errors during a split insertMany batch with [SPEC-898](https://jira.mongodb.org/browse/SPEC-898). I just need to come up with a format to express a very large string in a document (or consider writing this test in prose).